apps/backend/src/libs/modules/s3/base-s3.module.ts

```diff
@@ -0,0 +1,70 @@
+import {
+	DeleteObjectCommand,
+	type DeleteObjectCommandOutput,
+	PutObjectCommand,
+	type PutObjectCommandOutput,
+	S3Client,
+} from "@aws-sdk/client-s3";
+
+import { type UploadedFile } from "~/modules/files/files.js";
+
+import { type CommandParameters } from "./libs/types/types.js";
+
+type Constructor = {
+	accessKeyId: string;
+	bucketName: string;
+	region: string;
+	secretAccessKey: string;
+};
+
+class BaseS3 {
+	private bucketName: string;
+	private s3Client: S3Client;
+
+	public constructor(settings: Constructor) {
+		this.s3Client = new S3Client({
+			credentials: {
+				accessKeyId: settings.accessKeyId,
+				secretAccessKey: settings.secretAccessKey,
+			},
+			region: settings.region,
+		});
+		this.bucketName = settings.bucketName;
+	}
+
+	public async deleteFile(
+		parameters: Pick<UploadedFile, "key">,
```

Comment:
```suggestion
		key: string,
```
Do not overcomplicate things